src/CurveEditModel.cpp

```
@@ -0,0 +1,199 @@
+
+CurveEditModel::CurveEditModel() : QObject(nullptr) {}
+
+std::string CurveEditModel::makeKey(const QString& skeleton,
+                                     const QString& anim,
+                                     const QString& bone,
+                                     const QString& channel,
+                                     double time)
+{
+    // Quantize time to ms so floating-point round-trip doesn't break the key.
+    // Same precision as the dope sheet's keyframe-tick comparisons.
+    std::ostringstream ss;
```

Cppcheck (2.20.0)

[error] There is an unknown macro here somewhere. Configuration is required. If Q_ENUM is a macro then please configure it.

(unknownMacro)